internal/controller/dependencyupdatecheck_controller.go

```diff
@@ -310,6 +328,7 @@ func (r *DependencyUpdateCheckReconciler) createPipelineRun(comp component.GitCo
 		return nil, err
 	}
 
+	resources = nil
```

Comment:
Is this intended? I thought `defer` ran just before `return`, so will there be anything to clean up?

Reply:
If the code can reach here, there was no error happened, means we should not delete any resources we have created in this function, so we assign `nil` to resource, then the defer func will have nothing to delete.

Reply:
whenever we created a resource, add it to `resources`, if any error happens, all these resources will not be owned by the pipelinerun, they should should be deleted. If no happen happens, don't delete them, because they will be used and owned by the pipelinerun, they will be deleted automatically when the pipelinerun is deleted, or garbage collected.